MarcusBarnes/mik
# Issue 403 - add Excel fetcher.

**Github issue**: #403

# What does this Pull Request do?

Adds a fetcher capable of reading Excel (.xlsx) files. This fetcher is a drop-in replacement for the Csv fetcher in Csv toolchains.

# What's new?

* Added "box/spout":"*" to composer.json
* Added src/fetchers/Excel.php class file
* Updated src/metadataparsers/mods/CsvToMods.php to use fetcher configured in .ini file instead of hard-coded 'Csv' fetcher class.
* Added PHP unit tests for this new fetcher, plus an input file to use in the tests (tests/assets/excel/sample_metadata.xlsx)
* Modified other Csv test files to explicitly use the Csv fetcher class.

# How should this be tested?

This change is testable. To test:

1. Check out the issue-403 branch.
1. Run `composer update` or equivalent on your system.
1. Run the PHPUnit tests: `phpunit --exclude-group inputvalidators --bootstrap vendor/autoload.php tests`. You should get 49 tests, 69 assertions.
1. When you check out master branch again, you probably should rerun `composer update`.

# Additional Notes

We'll need to document the new Excel fetcher. Consistent with #372 and #373, I'm thinking something like https://github.com/MarcusBarnes/mik/wiki/Cookbook:-Templated-Metadata-Parser plus a mention in all the Csv toolchain wiki pages as an allowed option for `[FETCHER] class`. Draft is at https://github.com/MarcusBarnes/mik/wiki/Cookbook:-Using-the-Excel-fetcher (toolchain docs not updated yet).

I've tested this new fetcher on Linux and Windows with CsvToMods and Templated metadata parsers, and appears to work as intended.

# Interested parties

@MarcusBarnes 

@whikloj this is an example of the 'testable' workflow I mentioned in #402.
